pkg/utils/utils.go

```diff
@@ -36,8 +41,86 @@ var (
 	ErrFileExists                        = "file exists"
 	ErrInvalidFilterList                 = "failed to get filter list"
 	ErrMissingFilter                     = "no active filter to detach"
+	LocalCache                           = make(map[string]Metadata)
+	LocalCacheMutex      sync.Mutex
 )
 
+type Metadata struct {
+	Name      string
+	Namespace string
+}
+
+func GetServiceIP(client *kubernetes.Clientset, serviceName, serviceNamespace string, log logr.Logger) (string, error) {
+	service, err := client.CoreV1().Services(serviceNamespace).Get(context.Background(), serviceName, metav1.GetOptions{})
+	if err != nil {
+		log.Info("helper: error getting service")
+		return "", err
+	}
+	return service.Spec.ClusterIP, nil
+}
+
+func UpdateLocalCache(newCache map[string]Metadata) {
+	LocalCacheMutex.Lock()
+	defer LocalCacheMutex.Unlock()
+	for ip := range LocalCache {
+		if _, found := newCache[ip]; !found {
+			delete(LocalCache, ip)
+		}
+	}
+
+	for ip, metadata := range newCache {
+		LocalCache[ip] = metadata
+	}
+}
+
+func GetPodMetadata(ip string) (string, string, error) {
+	LocalCacheMutex.Lock()
+	defer LocalCacheMutex.Unlock()
+	if metadata, exists := LocalCache[ip]; exists {
+		return metadata.Name, metadata.Namespace, nil
+	}
+	return "", "", fmt.Errorf("pod metadata not found for IP %s in current cache", ip)
+}
+
+func LogFlowInfo(log logr.Logger, message *string, nodeName, srcIP, srcN, srcNS string, srcPort int, destIP, destN, destNS string, destPort int, protocol, verdict string) {
+	switch {
+	case srcN == "" && srcNS == "": // if no metadata for source IP
```

Comment:
We might've to rearrange this flow. Let's say `destN` and `destNS` is empty as well, we will end up printing "" strings for `DN` and `DNS`. Same applies to rest of the combinations below...

Also, why are we using `SHOSTIP` for Source IP field here?

Reply:
If it is a host networking pod such as aws-node pods or kube-proxy pods, we agreed to just say state that the IP is a host IP with no name or namespace.